### ISSUE TITLE
Umbrella: Idempotency improvements

### ISSUE THREAD

**Issue by detiber** [kubernetes-sigs/cluster-api-provider-aws#274]
/kind feature

General issue to track various idempotency improvements needed

- Consistent handling of resource idempotency
- Ensuring proper cleanup of resources if rollback is needed (especially in the clusterctl scenario, since it currently orphans resources if it encounters an issue)
- Verify consistent tag use
- Leverage client token support where possible

/priority important-soon

**Comment by vincepri** [comment]
I believe the only thing left in this umbrella is:

```
* Ensuring proper cleanup of resources if rollback is needed (especially in the clusterctl scenario, since it currently orphans resources if it encounters an issue)
```

@detiber Is there any particular way that this should be working? With the bootstrap model I think this issue should probably be solved upstream.

**Comment by detiber** [comment]
@vincepri I'm not sure that it is an upstream concern, this has to do more with making sure we are doing our best to clean up orphaned resources, especially in the case where resources cannot be tagged on create.

**Comment by vincepri** [comment]
Got it, so it should be fixed in #539. I was thinking about the wrong thing here, which was the cleanup in case of a rollback in clusterctl.

**Comment by ncdc** [comment]
I think we can close this out and use #539 going forward. Let's open new issues as needed.
